app/src/main/java/com/umc/anddeul/home/HomeFragment.kt

```diff
@@ -350,70 +367,117 @@ class HomeFragment : Fragment(), ConfirmDialogListener {
-            }
-        } else { // 안드로이드 SDK가 33보다 낮은 경우
-            if (permissionReadExternalGranted) {
-                // 이미 권한이 허용된 경우 해당 코드 실행
-                val postUploadActivity = Intent(activity, PostUploadActivity::class.java)
-                startActivity(postUploadActivity)
+    override fun onAcceptClicked() {
+        loadMemberList()
+    }
+
+    override fun onCancelClicked() {}
+
+    @SuppressLint("IntentReset")
+    fun startAlbumLauncher() {
+        val albumIntent = Intent(Intent.ACTION_PICK, MediaStore.Images.Media.EXTERNAL_CONTENT_URI)
+
+        // 이미지 여러개 선택 가능
+        albumIntent.putExtra(Intent.EXTRA_ALLOW_MULTIPLE, true)
+        // 액티비티를 실행한다.
+        albumLauncher.launch(albumIntent)
+    }
+
+    fun startPhotoPicker() {
+        pickMultipleMedia.launch(PickVisualMediaRequest(ActivityResultContracts.PickVisualMedia.ImageOnly))
+    }
+
+    // 포토피커 설정
+    fun settingPhotoPicker(imageUploadPossible: Int) {
+        pickMultipleMedia = registerForActivityResult(
+            ActivityResultContracts.PickMultipleVisualMedia(
+                imageUploadPossible
+            )
+        ) { uris ->
+            // 선택한 이미지들의 URI 목록을 처리하는 콜백
+            if (uris.isNotEmpty()) {
+                // 선택한 이미지가 있을 경우
+                val selectedImagesList = ArrayList(uris)
+
+                // 다음 액티비티로 전달할 intent 생성
+                val intent = Intent(requireContext(), PostWriteActivity::class.java)
+                intent.putParcelableArrayListExtra("selectedImages", selectedImagesList)
+
+                // 선택한 이미지들을 다음 액티비티로 전달
+                startActivity(intent)
             } else {
-                // 권한이 없는 경우 권한 요청
-                permissionLauncher.launch(android.Manifest.permission.READ_EXTERNAL_STORAGE)
+                // 선택한 이미지가 없을 경우
             }
         }
     }
 
-    override fun onAcceptClicked() {
-        loadMemberList()
-    }
+    // 앨범 런처 설정
+    fun settingAlbumLauncher(imageUploadPossible: Int) {
+        // 앨범 실행을 위한 런처
+        val contract2 = ActivityResultContracts.StartActivityForResult()
+        albumLauncher = registerForActivityResult(contract2) {
+            // 사진 선택을 완료한 후 돌아왔다면
+            if (it.resultCode == AppCompatActivity.RESULT_OK) {
+                // 선택한 이미지의 경로 데이터를 관리하는 Uri 객체 리스트를 추출
+                val uriclip = it.data?.clipData
+                uriclip?.let { clipData ->
+                    for (i in 0 until uriclip!!.itemCount) {
+                        if (selectedImages.size >= imageUploadPossible) {
+                            Snackbar.make(
+                                binding.root,
+                                "사진 첨부는 최대 ${imageUploadPossible}장까지 가능합니다.",
+                                Snackbar.LENGTH_SHORT
+                            ).show()
+                            break
+                        } else {
+                            val uri = uriclip.getItemAt(i).uri
```

Comment:
let 을 사용해서 null-check 가 되었으니 `uriClip` 이 아닌 `clipData` 를 사용하면 돼요. 그럼 `!!` 연산자를 사용하지 않아도 돼요 ㅎㅎ 코드에서 해당 연산자를 보면 지우지 않고서는 잠을 잘 수 없을 정도로 훈련해야합니다.
```suggestion
                uriclip?.let { clipData ->
                    for (i in 0 until clipData.itemCount) {
                        if (selectedImages.size >= imageUploadPossible) {
                            Snackbar.make(
                                binding.root,
                                "사진 첨부는 최대 ${imageUploadPossible}장까지 가능합니다.",
                                Snackbar.LENGTH_SHORT
                            ).show()
                            break
                        } else {
                            val uri = clipData.getItemAt(i).uri
```

Reply:
공부하면 좋은 것
- let 함수의 내부 구현 (간단해용)
- lambda with receiver 키워드로 검색 및 훑어보기

Reply:
앗 그러네요..! let 함수와 lambda with receiver에 대해 더 공부해보겠습니다!